Sources/Public/Animation/LottieAnimationViewInitializers.swift

```diff
@@ -100,90 +125,195 @@ extension LottieAnimationView {
 
   /// Loads a Lottie animation from a .lottie file in the supplied bundle.
   ///
-  /// - Parameter filePath: The string name of the lottie file with no file
-  /// extension provided.
-  /// - Parameter bundle: The bundle in which the file is located.
-  /// Defaults to the Main bundle.
+  /// - Parameter filePath: The name of the lottie file without the lottie extension. EG "StarAnimation"
+  /// - Parameter bundle: The bundle in which the lottie is located. Defaults to `Bundle.main`
+  /// - Parameter subdirectory: A subdirectory in the bundle in which the lottie is located. Optional.
   /// - Parameter animationId: Animation id to play. Optional
   /// Defaults to first animation in file
+  @available(iOS 13.0, macOS 10.15, tvOS 13.0, *)
   public convenience init(
     dotLottieName name: String,
     bundle: Bundle = Bundle.main,
+    subdirectory: String? = nil,
     animationId: String? = nil,
     dotLottieCache: DotLottieCacheProvider? = DotLottieCache.sharedCache,
-    configuration: LottieConfiguration = .shared)
+    configuration: LottieConfiguration = .shared) async throws
```

Comment:
I'm not entirely sure how useful these `async` animation view initializers are, since in most use cases I expect you would rather do all of the view hierarchy setup in one pass and then further configure the animation view in the callback. Seems like it probably doesn't hurt to provide them though. Any thoughts?

Reply:
Close all

Reply:
Hmm I would say it's probably an anti-pattern to construct a view asynchronously since that would prevent adding the view to the view hierarchy until its animation has loaded. Has anyone requested these initializers? I think creating an internal `Task` that loads the animation would be the expectation, similar to how `AsyncImage` works

Reply:
Agreed, this doesn't seem like a good idea. Will remove it.